### DISCUSSION TITLE
RM520n-GL not reaching full potential, but did earlier?

### DISCUSSION THREAD

**Discussion by soulnull** [iamromulan/cellular-modem-wiki#18] [Help - Quectel] [answered]
Hello,

I'm banging my head into a wall here. When I first got it up and running on OpenWRT, I was hitting gigabit speeds no sweat.. A few hours later, it went down and the modem needed a reboot, and since then, I can't get it over about 300mbps. The RM502q sitting next to me is hitting gig+ still, so not a tower issue.

In all of my diagnosing, the only oddity I've found is that sirq on the modem is maxing out during a speed test, absolutely pegged.

A major issue is this is an outdoor enclosure install, and after 7 hours of physical installation today (not including the now 4 hours of configuration/misconfiguration) including battling multiple hornets nests in the enclosure, I really don't want to climb back up and take the whole thing apart to fetch it. RSRP is -70, RSRQ -3, SNR 34, 100mhz n41 PCC + 15mhz n71 SCC.

My suspicion is openwrt is having the modem do CLAT instead of the router itself.. I'm certainly not the smartest with networking and especially clat/ipv6 stuff, and would absolutely 100% appreciate any direction someone can point me in to figure out how I botched this.

OpenWRT is getting ipv4 DHCP of 192.0.0.2, gateway 192.0.0.1.. Looks fine. ipv6 address is also pulling, and the 464xlat interface does appear. I suspect it's clat related because initially, disabling the ipv4 interface bumped it up to about 450mbps, but maybe I'm being delusional or overthinking it.. I'm so tired and lost right now though. I hate asking for help, but I know when I'm out of luck, and I'm out of luck.

I very humbly thank you for any guidance you can provide.

**Comment by tutugreen** [comment]
How did you connect to the module?
m2@pcie
m2@usb(2.0,3.0?)
m2@pcie rc(like this notes)@ethernet ？

try (adb login or ssh login into the module) using iperf with the module or looping traffic test to see if it's a problem between the module and the router. using dmesg / top (in adb/ssh) to find obvious problem (quectel module do thermal throttle on some case) / cpu/mem usage.

also use AT+QRSRP to check signal per antenna (maybe lost some antenna attach).

as last move, you said it was normal before, so you can qfirehose flash it (be careful different hw version, ec / ap / aa), it will return to factory settings (I mean all apn,  ethernet, lock settings, back to normal etc.)

**Comment by soulnull** [comment]
Hey, thank you so much for jumping in. Was actually working on this yet again just as your reply came in.

qrsrp is -73,-67,-66,-65 (so many at commands that I can't remember them all, was looking for this one to check exactly the scenario you mentioned, thanks)

m.2@pcie via 2.5gbe rj45 - did all the steps outlined with great success in the guide on my Linux desktop, NetworkManager was truly plug and play, but openwrt is a much more finicky beast.

To further complicate, the router is on the ceiling in a service closet behind the fridge, so just hooking the network cable to the PC across the residence is also tricky. Would need to pull out the fridge and climb a ladder, then get a 40 foot cable to drape it to the PC.

I think the "it was normal before" is a red herring, a setting I set without resetting the modem that applied when i reset.. Openwrt wasnt pulling an ipv4 address at that point, only ipv6.. And 464xlat did the translation. Try as I might, I can't recreate whatever the hell set of parameters I had working correctly briefly. I had it pulling a gigabit on the router earlier today after removing 464xlat, but it wasnt passing traffic to the network. Was getting correct ping responses from both v4 and v6 on the router, but nothing on the network itself (occasionally I can ping but not pass traffic, another confusing phenomenon.. But both pings and traffic on the router was good, but neither on the network).. Then I reset openwrt to see if that'd get it going and lost even that.. So put it back how it currently "works", 464xlat with 4 and 6.. disabled ip passthrough for now, pulls the 192.168.225.# address.. It absolutely refused to work like that initially, but now its fine?? But still 300mbps with pegged sirq on the modem.. Maybe I'm just not cut out for this..

I ran iperftest and.... The oddities continue. Best I could get was about 120mbps... *less* than I get running speedtest on the module itself (about 180mbps), which is less than the network is pulling (300mbps)...uhhhhh.. what? The on-device speedtest has been consistent though, even when it was giving me 1350mbps down on desktop, on-device never could clear around 180mbps.. I just chalked that up to internal processing/resources being too limited... But iperf results are fairly inconsistent (between 80-120mbps) and less than I'm actually getting through it.. Either I'm truly losing my mind or that doesn't make sense.. Or both. Definitely possibly both.

qfirehose flash...... Ah, I know that should be an "easy" last resort, but... With openwrt connected to USB, the modem kept rebooting, so I just disconnected it entirely and am at the mercy of the network/tailscale.. So reflashing will involve climbing the ladder and pulling the enclosure apart again. I also notice flashing really doesn't reset most of my settings, which can be quite frustrating as there is no real way to truly factory reset these things. Most of my network/PCI related settings that I'd be suspicious of are generally the ones I notice aren't really affected.. But.. Maybe thats what needs done. I really don't know..

Thanks for trying to help though.. Truly.. I don't know if any of this was helpful for diagnosis, I lean more towards this being an openwrt issue in how it's interacting with the modules network connection, but clearly the modem plays a part.. Arrrgh.

**Comment by soulnull** [comment, marked as answer]
Alright, easy "fix", disabling ipv6 in the APN and forcing T-Mobile to do the CLAT seemed to resolve it. It's still giving an ipv6 address, but 464xlat interface never comes up and Im getting gig+ again.. So I'll close this. Hopefully this works long-term.
